src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1447,27 +1461,95 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
+                    // locals have slots. The AccessExpression of the BoundConditionalAccess is another BoundConditionalAccess, this time
+                    // with a receiver of the GetB() BoundCall. Attempting to get a slot for this receiver will fail, and we'll
+                    // return an array with just the slot for a.
+                    switch (operand.Kind)
+                    {
+                        case BoundKind.Conversion:
+                            // PROTOTYPE(NullableReferenceTypes): Detect when conversion has a nullable operand
+                            operand = ((BoundConversion)operand).Operand;
+                            continue;
+                        case BoundKind.ConditionalAccess:
+                            var conditional = (BoundConditionalAccess)operand;
+
+                            slot = MakeSlot(conditional.Receiver);
+                            if (slot > 0)
+                            {
+                                // If we got a slot we must have processed the previous conditional receiver.
+                                Debug.Assert(_lastConditionalAccessSlot == -1);
+
+                                slotListBuilder.Add(slot);
+
+                                // When MakeSlot is called on the nested AccessExpression, it will recurse through receivers
+                                // until it gets to the BoundConditionalReceiver associated with this node. In our override,
+                                // we substitute this slot when we encounter a BoundConditionalAccessReceiver, and reset the
+                                // _lastConditionalAccess field.
+                                _lastConditionalAccessSlot = slot;
+                                operand = conditional.AccessExpression;
+                                continue;
+                            }
+
+                            // If there's no slot for this receiver, there cannot be another slot for any of the remaining
+                            // access expressions.
+                            break;
+                        default:
+                            // Attempt to create a slot for the current thing. If there were any more conditional accesses,
+                            // they would have been on top, so this is the last thing we need to specially handle.
+
+                            // PROTOTYPE(NullableReferenceTypes): When we handle unconditional access survival (ie after
+                            // c.D has been invoked, c must be nonnull or we've thrown a NullRef), revisit whether
+                            // we need more special handling here
+
+                            slot = MakeSlot(operand);
+                            if (slot > 0)
+                            {
+                                // If we got a slot we must have processed the previous conditional receiver.
+                                Debug.Assert(_lastConditionalAccessSlot == -1);
```

Comment:
Is this `default` case handling `BoundConditionalAccess` / `BoundConditionalReceiver` or all other operands except those? If the latter, consider moving the `Debug.Assert` up before `MakeSlot`.

Reply:
The latter.

---
In reply to: [204187214](https://github.com/dotnet/roslyn/pull/28722#discussion_r204187214) [](ancestors = 204187214)

Reply:
@cston after thinking though the state changes, I'm moving this back into the conditional. In the case of `a?.ToString()`, we won't have processed the `BoundConditionalReceiver` from the `ToString` `BoundCall` by this point. It is correct as originally written.

---
In reply to: [204188950](https://github.com/dotnet/roslyn/pull/28722#discussion_r204188950) [](ancestors = 204188950,204187214)